### PR TITLE
Add Policy Plugins for Custom K-Rail Policies

### PR DESCRIPTION
**Pull request by somethingnew2-0**
Add the ability to make custom K-Rail Policy Plugins using GRPC over localhost and [Hashicorp's go-plugin](https://github.com/hashicorp/go-plugin) interface.

An example plugin is provided that can be used as a template for other plugins.

### CODE REVIEW

**Review by dustin-decker — DISMISSED** [review]
Awesome work, Peter :rocket:

**Comment by somethingnew2-0** [comment]
Ok, updated the README.md and bumped helm chart version.  This PR should be good to go.  Thanks for the review @dustin-decker 😄

File: charts/k-rail/Chart.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 name: k-rail
 description: Kubernetes security tool for policy enforcement
 home: https://github.com/cruise-automation/k-rail
-version: v3.1.0
+version: v3.2.0
```
Comment (somethingnew2-0) [inline charts/k-rail/Chart.yaml:5]: Do I do this now or in a follow-up PR?

Comment (dustin-decker) [reply charts/k-rail/Chart.yaml:5]: Now

**Comment by broamski** [comment]
Is plugin checksum validation something we should consider? Too paranoid?

**Comment by somethingnew2-0** [comment]
@broamski I'm not sure what plugin checksum validation would protect against as K-Rail does not have a backend database and separate administration control-plane to register the enabled plugin hashes unlike Vault.  If plugin checksums were required they would probably be listed in the K-Rail configuration file which could already be modified by anyone with privileges to deploy K-Rail + K-Rail plugin binaries.  Feels too paranoid to me.

**Review by shellw — APPROVED** [review]

**Pull request merged**